### PR TITLE
Include method of getting settings from the module

### PR DESCRIPTION
**Pull request by jas8522**
Perhaps there's an alternate internal undocumented function that does this? If not, here we go :)

### CODE REVIEW

File: addon-modules/hooks.md [hunk 1]
```diff
@@ -11,6 +11,33 @@ Hooks that the module should define within WHMCS are defined in a file named “
 This should be within your custom module folder.
 That will then become detected and any hooks that exist will be loaded on every page of WHMCS.
 
+Module settings are not passed to hook functions. If you require access to the module's settings, they will need to be retrieved from the database. Add this to the top of your hooks.php file under the opening PHP tag: 
+
+```use Illuminate\Database\Capsule\Manager as Capsule;```
```
Comment (MarkehMe) [inline addon-modules/hooks.md]: You could also do: 
`use WHMCS\Database\Capsule `

Which might be preferred?

File: addon-modules/hooks.md [hunk 2]
```diff
@@ -11,6 +11,33 @@ Hooks that the module should define within WHMCS are defined in a file named “
...
+ * - Associative array containing settings if no settingname provided
+ * - value of type stored in DB if settingname is provided 
+ */
+function getSettings($modulename, $settingname = null){
+	
+	if ($settingname === null){
+		return (array) Capsule::table('tbladdonmodules')->where('module', $modulename)->get();
+	}
+	else{
+		return Capsule::table('tbladdonmodules')->where( array(
```
Comment (MarkehMe) [inline addon-modules/hooks.md]: I think the better way for this is:
```php
Capsule::table("tbladdonmodules")
    ->where(... blah ...)
    ->first()
    ->value;
```

File: addon-modules/hooks.md (hunk 1 above)
Comment (jas8522) [inline addon-modules/hooks.md]: Definitely.

File: addon-modules/hooks.md (hunk 2 above)
Comment (jas8522) [inline addon-modules/hooks.md]: Indeed! I was only doing it this way in case people wanted to get multiple variables at a time :)

**Comment by mattpugh** [comment]
An even better approach for this would probably be:

```
$settings = Capsule::table('tbladdonmodules')
    ->where('module', 'project_management')
    ->pluck('value', 'setting');
```

By using `pluck` you'll get back an associative array that allows you to access settings using something like `$settings['namehere']`.

To include this in the native docs, it would also be important to see the example function name be namespaced/prefixed with the addon module name to avoid multiple hooks attempting to declare the same function and failing.

**Comment by mattpugh** [comment]
Since WHMCS 7.3 there has been a model for addon settings. This can be found at `\WHMCS\Module\Addon\Setting` and is the recommended way of obtaining values of addon module settings in custom hook files.  Documentation is available at https://docs.whmcs.com/classes/7.4/WHMCS/Module/Addon/Setting.html

**Comment by jas8522** [comment]
I'm glad to hear there's a model to access this directly now! But shouldn't there be something in the module Developer documentation describing how to use it?

**Pull request closed without merging**